core/src/main/java/org/mskcc/oncotree/utils/TumorTypesUtil.java

```diff
@@ -20,32 +26,22 @@
 /**
  * Created by Hongxin on 2/25/16.
  */
+@Component
 public class TumorTypesUtil {
-    private static final String PROPERTY_FILE = "classpath:application.properties";
-    private static List<String> TumorTypeKeys = Arrays.asList("code", "name", "nci", "level", "umls", "maintype", "color");
 
-    public static Map<String, TumorType> getTumorTypes() {
-        Map<String, TumorType> tumorTypes = new HashMap<>();
-        Properties properties = getProperties();
-        try {
-            Version version = VersionUtil.getVersion("realtime");
-            CacheUtil.resetMainTypesByVersion(version);
+    private final static Logger logger = Logger.getLogger(TumorTypesUtil.class);
 
-            tumorTypes = parseFromRaw(new FileInputStream(properties.getProperty("tumor_type_file_path")), version);
-        } catch (FileNotFoundException e) {
-            e.printStackTrace();
-        }
-        return tumorTypes;
-    }
+    private static OncoTreeRepository oncoTreeRepository;
+    @Autowired
+    public void setOncoTreeRepository(OncoTreeRepository property) { oncoTreeRepository = property; }
+
+    private static final String PROPERTY_FILE = "classpath:application.properties";
+    private static List<String> TumorTypeKeys = Arrays.asList("code", "name", "nci", "level", "umls", "maintype", "color");
```

Comment:
So here we are leaving the model used by the frontend intact and filling the (frontend) model from what we get from our repository. This is ok. I would recommend that if we have different models in the repository and in the web layer, that we do model translation in a service layer (rather than in the repository). We can move towards that in a future PR